scatter_gather.go

```diff
@@ -64,21 +65,12 @@ func (sg *scatterGather[_, R]) gather(ctx context.Context) <-chan R {
 			log.Debugw("Completed scatter gather", "elapsed", elapsed.String())
 		}()
 
-		var rerr error
-		ticker := time.NewTicker(sg.maxWait)
-		for {
+		for r := range sg.out {
 			select {
 			case <-ctx.Done():
-				return
-			case <-ticker.C:
-				log.Debugw("Maximum scatter gather wait time reached; closing off channels", "rerr", rerr)
-				return
-			case r, ok := <-sg.out:
-				if !ok {
-					log.Debugw("Gathered all scatters in time", "rerr", rerr)
-					return
-				}
-				gout <- r
+				continue
```

Comment:
Why `continue` on context done?

Reply:
because we still need to drain `sg.out` so that it doesn't just hang around forever.